### ISSUE TITLE
Can't use `container` as a property name of a model 

### ISSUE THREAD

**Issue by Dragna** [apple/swift-openapi-generator#695]
### Description

In our API, one of the model has a `container` property, this conflict with the `container` from the `decoder`.

<img width="1153" alt="Capture d’écran 2024-12-10 à 13 41 43" src="https://github.com/user-attachments/assets/d1c04da4-e09a-4d14-afdb-01ebb2093eff">

wouldn't using `self.` help in mitigating this problem ? 

### Reproduction

openapi-generator-config.yaml

```
generate:
  - types
  - client
accessModifier: public
```

### Package version(s)

- swift-openapi-generator : 1.5.0
- swift-openapi-runtime: 1.7.0
- swift-openapi-urlsession: 1.0.2

### Expected behavior

I can use `container` as a property of a model in my schema

### Environment

```
swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0
```

**Comment by simonjbeaumont** [comment]
That's unfortunate, let me take a look.

**Comment by czechboy0** [comment]
Yeah, it'd have to be added to this list: https://github.com/apple/swift-openapi-generator/blob/61ec6b61a1c0c1e0f96351cf6159b0d97c5b27f1/Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/SwiftSafeNames.swift#L73

Unfortunately that'd be considered a breaking change, if done without a feature flag.

I wonder if the nameOverrides in our active proposal would actually be a better solution: https://forums.swift.org/t/proposal-soar-0013-idiomatic-naming-strategy/76242/23 (But that'll still be a few more days before being available to use)

**Comment by simonjbeaumont** [comment]
I think we could do this without a breaking change. We shouldn't change the property name, but use explicit self.

**Comment by Dragna** [comment]
> We shouldn't change the property name, but use explicit self.

yes that was what I was implying in my previous post, I don't know the ins and outs of how the generator works, but adding an explicit `self.` from what I understand should do the trick as we would access the property of the object and not the local scoped variable.

Thank you for your quick answers and fix 🙂 .

and thanks for the work on the generator it will be really useful for our project 🙂 .

**Comment by simonjbeaumont** [comment]
> Thank you for your quick answers and fix 🙂 .
> 
> and thanks for the work on the generator it will be really useful for our project 🙂 .

No problem. If you have a moment to test out the branch (or main, if we've merged the PR by the time you read this), it'd be good to know it has resolved your issue.

**Comment by Dragna** [comment]
@simonjbeaumont 

I tested it, it's working fine for the decode method, but the encode method needs to have the explicit `self.` too.

![Capture d’écran 2024-12-11 à 09 03 50](https://github.com/user-attachments/assets/117547a1-cfc6-4e53-914d-e65fe7c30b7a)

**Comment by simonjbeaumont** [comment]
I thought it might need a broader pass, but wanted to unblock you quickly. I'll take a broader look at this later on and try and do something more systematic.

**Comment by simonjbeaumont** [comment]
Release: https://github.com/apple/swift-openapi-generator/releases/tag/1.5.1
